shef/shef_parser.py

```diff
@@ -0,0 +1,3858 @@
+                                    if self._reject_problematic :
+                                        return []
+                                    break
+                            if p > 0 :
+                                if not time_overrides[p] and last is not None and time_overrides[last] :
+                                    time_overrides[p] = time_overrides[last]
+                                if relativetime_overrides[p] is None and last is not None and relativetime_overrides[last] is not None :
+                                    relativetime_overrides[p] = relativetime_overrides[last]
+                            if not skip_parameter :
+
+                                if hdr_param_info[p].parameter_code[3] == 'F' and not hdr_param_info[p].createtime :
+                                    self.warning(f"Forecast parameter [{hdr_param_info[p].parameter_code}] value [{value}] does not have creation date")
+
+                                outrecs.append(hdr_param_info[p].get_output_record(
+                                    revised,
+                                    msg_source,
+                                    location,
+                                    time_overrides[p] if time_overrides[p] else last_explicit_time,
+                                    relativetime_overrides[p],
+                                    createtime_override_str,
+                                    units_override,
+                                    duration_unit if duration_unit != 'Z' else None,
+                                    duration_value if duration_unit != 'Z' else None,
+                                    value,
+                                    qualifier,
+                                    comment))
+
+                                outrec_pos += 1
+                                last = p
+                            else :
+                                skip_parameter = False
+                        value = None
+                        comment = None
+                        duration_unit = 'Z'
+                        duration_value = None
+                        p += 1
+                except ShefParser.Exc as e :
+                    self.error(exc_info(e))
+                    if self._reject_problematic :
+                        return []
+            if outrec_pos < param_count :
+                self.warning(f"Value count ({outrec_pos}) is less than parameter count ({param_count}) for location [{location}]")
+        return outrecs
+    
+def parse(
+        input_name: Optional[str],
+        output_name: Optional[str],
+        output_format: int,
+        append_output: bool,
+        log_name: Optional[str],
+        log_level: str,
+        append_log: bool,
+        log_timestamps: bool,
+        shefparm: Optional[str],
+        use_defaults: bool,
+        shefit_times: bool,
+        reject_problematic: bool,
+        loader_spec: str,
+        unload: bool,) :
```

Comment:
Could we provide defaults for some of these?  Entering values for all 14 arguments is pretty tedious when you a user be fine with the defaults for several/most of them.

Reply:
Sure.